typescript/cli/src/deploy/warp.ts

```diff
@@ -677,11 +677,12 @@ async function enrollRemoteRouters(
           strategyUrl,
         });
       const transactionReceipts = await submitter.submit(...mutatedConfigTxs);
-
-      return logGreen(
-        `✅ Router enrollment update successfully submitted with ${submitter.txSubmitterType} on ${chain}:\n\n`,
-        indentYamlOrJson(yamlStringify(transactionReceipts, null, 2), 4),
-      );
+      if (transactionReceipts && transactionReceipts.length > 0) {
+        return logGreen(
+          `✅ Router enrollment update successfully submitted with ${submitter.txSubmitterType} on ${chain}:\n\n`,
+          indentYamlOrJson(yamlStringify(transactionReceipts, null, 2), 4),
+        );
+      }
     }),
```

Comment:
does the `submit` return type allow for undefined? might make more sense to have the safe submitter return empty array or something

Reply:
I agree, seems like the bug wasn't caught early because the `yamlStringify` function accepts as the first argument a param of type `any` while the `submit` function might not return a value

Reply:
@xeno097 Yup, this was mostly due a change in the sdk version: https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4415

I haven't dug into it deeply, but i seems the downgraded version returns undefined instead of an array. We'll handle it. Another reason is that we're using a .js file for our gnosis safe service, so no typing to help.